lib/block-write-stream.ts

```diff
@@ -69,9 +113,9 @@ export class BlockWriteStream extends Writable {
 	/**
 	 * @summary Write buffered data before a stream ends, called by stream internals
 	 */
-	_final(callback: () => void) {
+	_final(callback: (error?: Error | void) => void) {
 		debug('_final');
-		this.__final().then(callback, this.destroy.bind(this));
+		asCallback(this.__final(), callback);
```

Comment:
`this.destroy` is never called now. But also previously the callback was never called in case of error.

Reply:
[According to the docs](https://nodejs.org/api/stream.html#stream_writable_final_callback) the final callback does take an error argument and should destroy the stream if that's present, so this should still work as it did before

Reply:
I've added the explicit destroy() back